$NetBSD: patch-ax,v 1.5 2008/04/03 07:59:08 tonnerre Exp $

Don't deadlock on exit with multiple X forwarded channels.
Don't use X11 port which can't be bound on all IP families.
Fixes CVE-2008-1483.

--- channels.c.orig	2007-06-25 09:04:47.000000000 +0000
+++ channels.c
@@ -2905,9 +2905,6 @@ x11_create_display_inet(int x11_display_
 				debug2("bind port %d: %.100s", port, strerror(errno));
 				close(sock);
 
-				if (ai->ai_next)
-					continue;
-
 				for (n = 0; n < num_socks; n++) {
 					close(socks[n]);
 				}
